File: examples/medium.js
function quicksort(arr) {
  if (arr.length <= 1) return arr;
  const pivot = arr[Math.floor(arr.length / 2)];
  const left = arr.filter((x) => x < pivot);
  const middle = arr.filter((x) => x === pivot);
  const right = arr.filter((x) => x > pivot);
  return [...quicksort(left), ...middle, ...quicksort(right)];
}

const arr = [3, 6, 8, 10, 1, 2, 1, 5, 7, 9, 4];
console.log('Исходный массив:', arr);
console.log('Отсортированный:', quicksort(arr));

File: examples/medium.go
package main

import (
	"fmt"
)

func quicksort(arr []int) []int {
	if len(arr) <= 1 {
		return arr
	}
	pivot := arr[len(arr)/2]
	left := []int{}
	middle := []int{}
	right := []int{}
	for _, x := range arr {
		if x < pivot {
			left = append(left, x)
		} else if x == pivot {
			middle = append(middle, x)
		} else {
			right = append(right, x)
		}
	}
	left = quicksort(left)
	right = quicksort(right)
	return append(append(left, middle...), right...)
}

func main() {
	arr := []int{3,6,8,10,1,2,1,5,7,9,4}
	fmt.Println("Исходный массив:", arr)
	fmt.Println("Отсортированный:", quicksort(arr))
} 